packages/amplify-category-function/src/__tests__/provider-utils/awscloudformation/utils/environmentVariablesHelper.test.ts

```diff
@@ -85,11 +82,198 @@ describe('ensureEnvironmentVariableValues', () => {
+  ])(
+    'prompts the user $numSelectFunctionCalls times when the first question is answered with $answer',
+    async ({ numCarryUpdateCalls, answer, numSelectFunctionCalls: numSelectFunctionCalls }) => {
+      prompterMock.pick.mockResolvedValueOnce(answer);
+      prompterMock.pick.mockResolvedValueOnce(abortKey);
+
+      await envVarHelper.askEnvironmentVariableCarryOrUpdateQuestions(context, envName, false);
+
+      expect(prompterMock.pick).toBeCalledTimes(numCarryUpdateCalls + numSelectFunctionCalls);
+    },
+  );
+
+  it('prompts the user with a specific question and choices when update is selected', async () => {
+    prompterMock.pick.mockResolvedValueOnce('update');
+    prompterMock.pick.mockResolvedValueOnce(abortKey);
+
+    await envVarHelper.askEnvironmentVariableCarryOrUpdateQuestions(context, envName, false);
+
+    expect(prompterMock.pick).toHaveBeenCalledTimes(2);
+    expect(prompterMock.pick).toHaveBeenNthCalledWith(
+      1,
+      'You have configured environment variables for functions. How do you want to proceed?',
+      howToProceedChoices,
+    );
+    expect(prompterMock.pick).toHaveBeenNthCalledWith(2, 'Select the Lambda function you want to update values', selectFunctionChoices);
+  });
+
+  it('prompts the user with a specific question and choices when function is selected', async () => {
+    prompterMock.pick.mockResolvedValueOnce('update');
+    prompterMock.pick.mockResolvedValueOnce('testFunc');
+    prompterMock.pick.mockResolvedValueOnce(abortKey); // select variable
+    prompterMock.pick.mockResolvedValueOnce(abortKey); // select function
+
+    await envVarHelper.askEnvironmentVariableCarryOrUpdateQuestions(context, envName, false);
+
+    expect(prompterMock.pick).toHaveBeenCalledTimes(4);
+    expect(prompterMock.pick).toHaveBeenNthCalledWith(
+      1,
+      'You have configured environment variables for functions. How do you want to proceed?',
+      howToProceedChoices,
+    );
+    expect(prompterMock.pick).toHaveBeenNthCalledWith(2, 'Select the Lambda function you want to update values', selectFunctionChoices);
+    expect(prompterMock.pick).toHaveBeenNthCalledWith(
+      3,
+      "Which function's environment variables do you want to edit?",
+      selectVariablesChoices,
+    );
+  });
+
+  it('prompts the user to provide a new variable value when a variable is selected', async () => {
+    prompterMock.pick.mockResolvedValueOnce('update');
+    prompterMock.pick.mockResolvedValueOnce('testFunc');
+    prompterMock.pick.mockResolvedValueOnce('testEnv');
+    prompterMock.input.mockResolvedValueOnce('testVal');
+    prompterMock.pick.mockResolvedValueOnce(abortKey);
+    prompterMock.pick.mockResolvedValueOnce(abortKey);
+
+    await envVarHelper.askEnvironmentVariableCarryOrUpdateQuestions(context, envName, false);
+
+    // Expecting anything for optional prompter parameters.
+    // TODO: figure out how to better expect how this function is called.
```

Comment:
nit: Should this TODO be addressed? If not, please remove the comment.

Reply:
It feels important for it to be addressed, but I couldn't figure out a way to expect optional parameters that take functions as arguments. I can still remove the comment, unless you know of a way to do this.